src/test/java/org/kairosdb/core/properties/PropertiesValidatorTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Dropbox
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kairosdb.core.properties;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.name.Names;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.kairosdb.core.exception.KairosDBException;
+import org.kairosdb.core.reporting.MetricReporterService;
+import org.kairosdb.datastore.cassandra.CassandraConfiguration;
+
+import java.util.Optional;
+
+/**
+ * Tests for {@link PropertiesValidator}.
+ *
+ * @author Max Wittek (https://github.com/wittekm)
+ */
+public final class PropertiesValidatorTest {
+    private static final int SOME_POSITIVE_INT = 1234;
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Test
+    public void testWhenNothingSpecifiedValidatorPasses() throws KairosDBException {
+        ValidatorArgs args = new ValidatorArgs();
+
+        validatorWithInjection(args).validate();
+    }
+
+    @Test
+    public void testWhenOnlyAlignIsOnWithoutReporterTtlValidatorFails() throws KairosDBException {
+        exceptionRule.expect(KairosDBException.class);
+        exceptionRule.expectMessage("you must set kairosdb.reporter.ttl to a positive value");
+
+        ValidatorArgs args = new ValidatorArgs();
+        args.alignDatapointTtlWithTimestamp = Optional.of(true);
+
```

Comment:
Maybe I'm missing something but wouldn't the default TTL of 0 pass validation when an explicit positive number isn't specified?

Reply:
Nvm, I was confused. 0 is not allowed